lib/TH/generic/THStorage.h

```diff
@@ -28,6 +29,8 @@ typedef struct THStorage
     char flag;
     THAllocator *allocator;
     void *allocatorContext;
+    int type;
```

Comment:
Sorry, I deleted my comment as Clement wrote a comment whilst I was writing mine and it didn't seem valid any more.

In a generic file (i.e. one that uses THStorage macros), the macro "real" expands to the element type of the Storage, e.g. float, double, int, etc. See [here](https://github.com/torch/torch7/blob/master/lib/TH/THGenerateAllTypes.h#L5) for more details. 
i.e.
```C
THStorage *storage = luaT_checkudata(L, 1, torch_Storage);
real item; // This is a float if storage is a THFloatStorage, an int if it's a THIntStorage, etc.
```

Alternatively `#ifdef TH_REAL_IS_BYTE` or similar (defined in the same header) could probably help.

Just to be clear, (sorry if you know all this) THStorage is not an actual type, but part of the macro templating system.